st2client/st2client/commands/auth.py

```diff
@@ -70,6 +73,65 @@ def run_and_print(self, args, **kwargs):
                               attributes=self.display_attributes, json=args.json, yaml=args.yaml)
 
 
+class LoginCommand(resource.ResourceCommand):
+    display_attributes = ['user', 'token', 'expiry']
+
+    def __init__(self, resource, *args, **kwargs):
+
+        kwargs['has_token_opt'] = False
+
+        super(LoginCommand, self).__init__(
+            resource, kwargs.pop('name', 'create'),
+            'Authenticate user, acquire access token, and update CLI config directory',
+            *args, **kwargs)
+
+        self.parser.add_argument('username',
+                                 help='Name of the user to authenticate.')
+
+        self.parser.add_argument('-p', '--password', dest='password',
+                                 help='Password for the user. If password is not provided, '
+                                      'it will be prompted.')
+        self.parser.add_argument('-l', '--ttl', type=int, dest='ttl', default=None,
+                                 help='The life span of the token in seconds. '
+                                      'Max TTL configured by the admin supersedes this.')
+
+    def run(self, args, **kwargs):
+
+        if not args.password:
+            args.password = getpass.getpass()
+        instance = self.resource(ttl=args.ttl) if args.ttl else self.resource()
+
+        manager = self.manager.create(instance, auth=(args.username, args.password), **kwargs)
+
+        cli = BaseCLIApp()
+        logging.basicConfig(
+            level=logging.CRITICAL,
+            handlers=[logging.NullHandler]
+        )
+        cli.LOG = logging.getLogger("st2client.base")
+        cli._cache_auth_token(token_obj=manager)
+
+        # Update existing configuration with new credentials
+        config_file = "%s/.st2/config" % expanduser("~")
+        config = ConfigParser()
+        config.read(config_file)
```

Comment:
No exception is raised, just an empty list is returned.

```
(Pdb) config.read("/tmp/foodoesntexist.txt")
[]
```

This just means there wasn't a file there to populate the config object.

If I run this with the real code, and the config file already exists, you see this:

```
(Pdb) config.read(config_file)
['/home/vagrant/.st2/config']
```

Either way, we're always overwriting the two parameters in the "credentials" section, and writing it to disk.